### PR TITLE
PML-157: bugfix -  unnormalized probabilities in sloscomputegraph

### LINKED ISSUE #111: Unnormalized probabilities in `SLOSComputeGraph`.
**Describe the bug**
`SLOSComputeGraph.compute_probs_from_amplitudes` is returning unnormalized probabilities.
This leads to errors in `SLOSComputeGraph.compute_probs` which in turn leads to errors in `FidelityKernel`.

**To Reproduce**
```python
import torch
from perceval import BS
from merlin import build_slos_distribution_computegraph

unitary = torch.tensor(BS().compute_unitary()).unsqueeze(0)
unitary = unitary.to(torch.complex64)

slos_graph = build_slos_distribution_computegraph(m=2, n_photons=2)

# Calculate probs directly from amplitudes
_, amplitudes = slos_graph.compute(unitary, [1, 1])
_, probs2 = slos_graph.compute_probs_from_amplitudes(amplitudes)

print("The probs for a beamsplitter are:", probs2)
```
```
The probs for a beamsplitter are: tensor([1.0000, 0.0000, 1.0000])
```

### PR DESCRIPTION
**Pull request by CassNot**
## Summary

This PR solves the issue https://github.com/merlinquantum/merlin/issues/111
When apply `graph.compute` then `graph.compute_probs_from_amplitudes`, the probabilities were scaled twice by the `self.norm_factor_input` and `self.norm_factor_output` and this was causing the error raised.
I added two tests to reproduce the error and removed the dual scaling in `compute_probs_from_amplitudes` as it is necessary called after `graph.compute()`.
The scaling in compute is preserved.

## Related Issue
   
Related Jira: PML-157

## Type of change
- [x] Bug fix

## How to test / How to run

1. Command lines

```
pytest tests/pcvl_pytorch/test_slos_computegraph.py
pytest
```

## Checklist
- [x] PR title includes Jira issue key (e.g., PML-126)
- [x] "Related Jira ticket" section includes the Jira issue key (no URL)
- [x] Code formatted (ruff format)
- [x] Lint passes (ruff)
- [x] Static typing passes (mypy) if applicable
- [x] Unit tests added/updated (pytest)
- [x] Tests pass locally (pytest)
- [x] Tests pass on GPU (pytest)
- [x] Test coverage not decreased significantly
- [x] Docs build locally if affected (sphinx)
- [x] Dependencies updated (if needed) and pinned appropriately
- [x] PR description explains what changed and how to validate it

### CODE REVIEW

**Comment by CassNot** [comment]
I implemented the single-truth helper in merlin/utils/normalization

Now, 
- `_renormalize_distribution_and_amplitudes` calls the shared `normalize_probabilities_and_amplitudes` which used a shared `probabilities_from_amplitudes` and `normalize_probabilities` (computation space dependent)
- compute_probs_from_amplitudes calls `probabilities_from_amplitudes` and `normalize_probabilities` while preserving slos_torchscript knowledge

I did not modify other normalization logic (in `feed_forward` or `statevector` for instance) as the normalization logic is different and can rely on branches or different object

**Review by jsenellart — APPROVED** [review]
